gradle-plugins/biz.aQute.bnd.gradle/src/main/java/aQute/bnd/gradle/AbstractBndrun.java

```diff
@@ -109,12 +112,26 @@ public RegularFileProperty getBndrun() {
 	 * @return The bundles to be added to a FileSetRepository for non-Bnd
 	 *         Workspace builds.
 	 */
-	@InputFiles
-	@PathSensitive(NONE)
+	@Internal
 	public ConfigurableFileCollection getBundles() {
 		return bundles;
 	}
 
+	/**
+	 * Wrapper of the {@link #getBundles()} method to allow <code>@Classpath</code> normalization and sorting.
+	 *
+	 * @return The sorted bundles.
+	 */
+	@Classpath
+	public Provider<List<File>> getBundlesSorted() {
+		return getBundles().getElements().map(
+			c -> c.stream()
+				.map(FileSystemLocation::getAsFile)
+				.sorted(Comparator.comparing(File::getName))
```

Comment:
We need to use a better comparator. 

```suggestion
				.sorted(IO.fileComparator(File::getAbsolutePath))
```

This will provide a consistent reproducible ordering over the complete path. See https://github.com/bndtools/bnd/commit/3d5359e677b6317afe1347a8d4b75e8be6f8f13b.